subt_ign/src/GameLogicPlugin.cc

```diff
@@ -2748,3 +2762,33 @@ double GameLogicPluginPrivate::FloorMultiple(double _n, double _m)
     out -= _m;
   return out;
 }
+
+/////////////////////////////////////////////////
+void GameLogicPluginPrivate::OnKineticEnergyEvent(
+    const ignition::msgs::Double &/*_msg*/,
+    const transport::MessageInfo &_info)
+{
+  ignition::msgs::Time localSimTime(this->simTime);
+  std::vector<std::string> topicParts = common::split(_info.Topic(), "/");
+  std::string name = "_unknown_";
+
+  // Get the name of the model from the topic name, where the topic name
+  // look like '/model/{model_name}/kinetic_energy'.
+  if (topicParts.size() > 1)
+    name = topicParts[1];
+
+  {
+    std::lock_guard<std::mutex> lock(this->eventCounterMutex);
+    std::ostringstream stream;
+    stream
+      << "- event:\n"
+      << "  id: " << this->eventCounter << "\n"
+      << "  type: KineticEnergyThreshold Surpassed\n"
```

Comment:
Recommend renaming the event type 
```suggestion
      << "  type: collision\n"
```